### ISSUE TITLE
esttex "drop" argument too aggressive

### ISSUE THREAD

**Issue by grantmcdermott** [lrberge/fixest#11]
Subtitle: Ignore variables with dictionary alias

Hi Laurent,

I've been trying to export a subset of my regression results where (a) I'm only really interested in some interaction terms, and (b) I further wish to rename these interaction terms with a dictionary alias. My problem is that in trying to drop the parent terms using the "drop" argument of `esttex()`, I invariably end of excluding the interactions variables of interest too... even when I have assigned them a dictionary alias. 

An example may help to better illustrate:

``` r
library(fixest)

aq = airquality
aq$Month = factor(aq$Month)

mod = feols(Ozone ~ Month / Wind, data = aq)
#> NOTE: 37 observations removed because of NA values (Breakup: LHS: 37, RHS: 0).

myDict = c("Month5:Wind"="May * Wind", "Month6:Wind"="Jun * Wind", 
           "Month7:Wind"="Jul * Wind", "Month8:Wind"="Aug * Wind",
           "Month9:Wind"="Sep * Wind")

## Keep everything, including parent terms (but rename interaction terms)
esttex(mod, dict = myDict)
#> \begin{table}[htbp]\centering
#> \caption{no title}
#> \begin{tabular}{lc}
#>  & \tabularnewline
#> \hline
#> \hline
#> Dependent Variable:&Ozone\\
#> Model:&(1)\\
#> \hline
#> \emph{Variables}\tabularnewline
#> (Intercept)&50.748$^{***}$\\
#>   &(15.748)\\
#> Month6&-41.793\\
#>   &(31.148)\\
#> Month7&68.296$^{***}$\\
#>   &(20.995)\\
#> Month8&82.211$^{***}$\\
#>   &(20.314)\\
#> Month9&23.439\\
#>   &(20.663)\\
#> May * Wind&-2.3681$^{*}$\\
#>   &(1.3162)\\
#> Jun * Wind&1.6825\\
#>   &(2.1141)\\
#> Jul * Wind&-7.0314$^{***}$\\
#>   &(1.5399)\\
#> Aug * Wind&-8.5224$^{***}$\\
#>   &(1.4015)\\
#> Sep * Wind&-4.2418$^{***}$\\
#>   &(1.2575)\\
#> \hline
#> \emph{Fit statistics}&  \\
#> Observations& 116\\
#> R$^2$ & 0.54732\\
#> Adjusted R$^2$ & 0.50889\\
#> \hline
#> \hline
#> \multicolumn{2}{l}{\emph{Normal standard-errors in parentheses. Signif Codes: ***: 0.01, **: 0.05, *: 0.1}}\\
#> \end{tabular}
#> \end{table}

## Try to drop only parent terms, but ends up excluding everything (including
## interactions renamed with a dictionary alias) 
esttex(mod, dict = myDict, drop = paste0("Month", 5:9))
#> \begin{table}[htbp]\centering
#> \caption{no title}
#> \begin{tabular}{lc}
#>  & \tabularnewline
#> \hline
#> \hline
#> Dependent Variable:&Ozone\\
#> Model:&(1)\\
#> \hline
#> \emph{Variables}\tabularnewline
#> (Intercept)&50.748$^{***}$\\
#>   &(15.748)\\
#> \hline
#> \emph{Fit statistics}&  \\
#> Observations& 116\\
#> R$^2$ & 0.54732\\
#> Adjusted R$^2$ & 0.50889\\
#> \hline
#> \hline
#> \multicolumn{2}{l}{\emph{Normal standard-errors in parentheses. Signif Codes: ***: 0.01, **: 0.05, *: 0.1}}\\
#> \end{tabular}
#> \end{table}
```

<sup>Created on 2019-12-29 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

I know that in the vignette you [write](https://cran.r-project.org/web/packages/fixest/vignettes/fixest_walkthrough.html#151_an_elaborate_example): _"Note that the actions performed by the arguments drop or order are performed before the renaming takes place with the argument dict."_

However, I'm wondering if it is possible to switch things around so that drop respects the assigned aliases? Failing that, is there a way to adjust drop's regexp so that it isn't so aggressive (such that it would allow me to keep the child interaction terms in the above example)?

PS &mdash; I have a second question about the dict argument, but I'll file that as a separate issue shortly.

**Comment by lrberge** [comment]
Hi Grant,

First of all: Happy new year!

Thanks for your comment! Having the 'drop' or 'order' arguments to be applied post aliasing looks desirable indeed. Sold! I'll implement it :-). 
Since I think applying it pre-aliasing can still be interesting, I'll make post-aliasing the default and add another argument for pre-aliasing (whose name is to be defined -- eg like drop.pre, order.pre).

More specific on your problem, you can solve it directly with regular expressions, by using: 
`drop = paste0("Month", 5:9, "$")`
where the dollar sign refers to the end of the character string.

Now I'm jumping to your second issue :-)

**Comment by lrberge** [comment]
Sorry: clicked on the wrong button -_-

**Comment by lrberge** [comment]
Just a comment: I initially implemented regular expressions because they are supa dupa powerful, and you can get rid of a lot of variables at once. 

In your case, to obtain the behavior that you want, you could use:
 `drop = "Month[[:digit:]]$"`
and it will get rid of only the variables equal to `Month6` to `Month9` -- by just using one character string. (But the drawback with regular expressions is that it's not really intuitive.) 

In any case, I'll still implement your suggestion of applying `drop` post-aliasing -- I fully agree it makes more sense that way.

**Comment by grantmcdermott** [comment]
Super, thanks Laurent. My actual use case is a little more complicated than this MWE but you've inspired me to be more precise in my regexp for the moment.

By the way, I just returned from the ASSA meetings. I was enthusiastically endorsing **fixest** to numerous people there, so hopefully you continue seeing an uptake in use!

**Comment by lrberge** [comment]
Fantastic, thanks for spreading the word! But now I have to keep up :-D

**Comment by lrberge** [comment]
Hi Grant, I've implemented your suggestions, namely: drop used post aliasing, and two variables with the same alias appear in the same row.

By the way, I've added:
 - that variables in interactions are reordered when needed, not to create two variables look different when they are the same (e.g. `x1:x2` and `x2:x1` should appear as the same variable)
 - aliases also apply to variables inside interactions (for example if `dict = c(x1="Wind", x2="Rain")` and you have the following variables in the estimation: x1, x2 and x1:x2, then you'll get the following aliases: `Wind`, `Rain` and `Wind $\times$ Rain`). I combined the interactions with `$\times$` to make it look nice in Latex, but the user can change it with the `interaction.combine` argument. Of course you can still rename the interaction directly in the dictionary. 

Thanks again for the comments!

**Comment by grantmcdermott** [comment]
Perfect. Thanks Laurent!

PS &mdash; I dig the new `interaction.combine` argument. One minor suggestion: Leave this on automatically, so that we get the `\times` symbol for interaction terms even when no dictionary is supplied. (They can always switch it off too.)

**Comment by lrberge** [comment]
Suggestion accepted! 

By the way, thanks to your MWE, I've updated the documentation with the `airquality` data set and it's much clearer now!

Might be of interest to you: I've added a new feature when dropping/ordering the variables. Now if you use an exclamation mark, it reverses the regular expression. For example: `drop = "Wind"` drops all variables containing "Wind", while `drop = "!Wind"` drops all variables **not** containing "Wind".
